sdks/go/pkg/beam/core/runtime/xlangx/expansionx/download.go

```diff
@@ -375,3 +380,68 @@ func jarExists(jarPath string) bool {
 	_, err := os.Stat(jarPath)
 	return err == nil
 }
+
+func getPythonVersion() (string, error) {
+	for _, v := range []string{"python", "python3"} {
+		cmd := exec.Command(v, "--version")
+		if err := cmd.Run(); err == nil {
+			return v, nil
+		}
+	}
+	return "", fmt.Errorf("no python installation found")
+}
+
+// SetUpPythonEnvironment sets up the virtual ennvironment required for the
+// Apache Beam Python SDK to run an expansion service module.
+func SetUpPythonEnvironment(extraPackage string) (string, error) {
+	py, err := getPythonVersion()
+	if err != nil {
+		return "", fmt.Errorf("no python installation found: %v", err)
+	}
+	extraPackages := []string{}
+	if len(extraPackage) > 0 {
+		extraPackages = strings.Split(extraPackage, " ")
+	}
+
+	// create python virtual environment
+	sort.Strings(extraPackages)
+	beamPackage := fmt.Sprintf("apache_beam[gcp,aws,asure,dataframe]==%s", core.SdkVersion)
+	cacheDir := "~/.apache_beam/cache"
```

Comment:
Put this up by the java constants,a nd build the javaJAR cache from this prefix, so we don't end up splitting our cache into different places accidentally.